### DISCUSSION TITLE
Wright Fisher forward simulation

### DISCUSSION THREAD

**Discussion by NiharaDeSilva** [molpopgen/fwdpy11#1335] [General]
I am trying to use fwdpy11 to simulate a population with the Wright Fisher forward in time simulator and output SNP data. 
So my questions are
whether fwdpy11 supports outputing SNP data?
Whether I could provide the effective population size as an input parameter?
Also whether it supports the discrete generation feature where i could simulate the populations for a given duration range(eg: 2,8). 
Appreciate your input. Thanks in advance.

**Comment by molpopgen** [comment]
Hi @NiharaDeSilva,

Welcome!  The answer to all of this is "yes', except for the duration range.  Assuming that you are referring to time intervals moving FORWARDS could simulate from [0, 8] but not from [2, 8] because it is unclear what the state of the population would be 2 generations into the future.

I will note that by "output SNP data", the output is a tskit TreeSequence and that 0 = ancestral, 1 = derived. I'll refer you to the manual for details.

All of these details are in the manual, which is linked to from the README, pypi, etc..

**Reply by NiharaDeSilva** [reply]
Thank you @molpopgen. Really appreciate your response. Going through the documentation i did not come across a straightforward method/function in fwdpy11 to simulate populations using the wright fisher model. Only a tskit method which outputs a tree sequence which is discussed under this link https://tskit.dev/tutorials/forward_sims.html.
Is this the only way, or is there a built-in function in fwdpy11 to simulate wright-fisher forward simulations?

**Reply by molpopgen** [reply]
There is such a function and there are many examples in the manual.  One such example is [here](https://molpopgen.github.io/fwdpy11/short_vignettes/evolvets_vignette.html), but there are lots.  The examples contain calls to functions provided by the package.  You can search for them in the manual.

fwdpy11 ONLY simulates Wright-Fisher models.  I'd suggest going through the various sections of the manual, starting at the beginning, to see how you specify the genetics, demography, recorder callbacks, etc..

Your link to tskit is a tutorial for that tool and not part of the fwdpy11 package.

**Reply by NiharaDeSilva** [reply]
tskit tree sequence outputs the nucleotides through the halpotypes() function [here](https://tskit.dev/tutorials/popgen.html). Is this supported in fwdpy11 tree sequence as well? Are there any limitations with fwdpy11 tree sequence.

**Reply by molpopgen** [reply]
The output from a fwdpy11 simulation is a tskit.TreeSequence via the function to convert populations to tree sequences.

I am going to close this discussion now as it has become open-ended and all of the questions asked have section titles in the manual that would answer them.  For example, one of the sidebar items is "Exporting data to tskit".
